### ISSUE TITLE
Tensorboard Integration

### ISSUE THREAD

**Issue by xanderdunn** [Kaixhin/FGLab#21]
FGLab has some built-in graph UI:
![experiment](https://cloud.githubusercontent.com/assets/1313618/18405729/8769aa20-76a9-11e6-8a5b-2d91138c2437.png)

TensorFlow's TensorBoard has a full-featured, interactive web GUI for visualizing training output:
![687474703a2f2f72757373656c6c73737465776172742e636f6d2f732f74656e736f72626f782f74656e736f72626f6172645f6c6f73732e706e67](https://cloud.githubusercontent.com/assets/1313618/18405825/61b02eac-76aa-11e6-9db7-aa85723e1ccd.png)

Would it be possible to store the [TensorBoard summary files](https://www.tensorflow.org/versions/r0.10/how_tos/summaries_and_tensorboard/index.html) in FBLab's database and then link from the FBLab web UI to open the TensorBoard UI?  Ideally, FBLab would even be able to read stored values (train loss, val loss, etc.) directly from the TensorBoard summary files.

This would be an awesome integration for TensorFlow users.

**Comment by Kaixhin** [comment]
If there are summary/log files being stored in a directory then FGLab will certainly store them in MongoDB's GridFS - but as individual files rather than the entire directory, so you would have to download all files one by one/utilise the API to get them via a script.

It's possible to add [custom fields with text](https://github.com/Kaixhin/FGMachine#experiments), and I could implement extra processing to recognise HTTP links and make them actual links, but this would require the aforementioned script to have been run, and TensorBoard to have been started.

For FGLab to try and produce its own graphs, you would have to have something parse the files written by the `SummaryWriter` in a different directory and place the JSON graphs into the experiment directory to be uploaded to FGLab.

Given TensorBoard's wealth of features and tight integration with TensorFlow, how exactly are you hoping to use FGLab with TensorFlow?

**Comment by xanderdunn** [comment]
I'm hoping to use FBLab to store and present experiment results.  

Tensorboard is a nice UI for viewing the results of a specific experiment, but it offers no solution to storing or viewing the results of a myriad of parameters.  Tensorboard doesn't do this:
![experiments](https://cloud.githubusercontent.com/assets/1313618/18414960/59fba7a0-7794-11e6-846e-b36c6976b93a.png)

This goal could be achieved without any kind of FGLab<->Tensorboard integration.  I think it's purely a matter of convenience, as opposed to needing to find the TensorBoard summary files manually on disk and get to it externally from the FGLab view.

**Comment by Kaixhin** [comment]
Option 1, which is to present TensorBoard graphs natively in FGLab, would require parsing the files somewhere - probably on the server. It _should_ be possible, but I don't have the resources to work on that.

Option 2 is to devise a script that makes it easy to download the files for TensorBoard to launch, and document it in the [examples](https://github.com/Kaixhin/FGLab/tree/master/examples). This is something I can tackle once I have more time. Let me know if you want to try either option yourself and I can answer questions about the FGLab code.

**Comment by Kaixhin** [comment]
Just a quite note to say that this is _very low priority_, i.e. I'm very unlikely to get around to this myself, but if you want to attempt it I'll do my best to answer questions.
